src/network/hobbits/hobbitsConnectionHandler.ts

```diff
@@ -0,0 +1,262 @@
+      }, this.requestTimeout);
+      this.once(`response ${id}`, (err, data) => {
+        clearTimeout(timeout);
+        delete this.requests[id];
+        if (err) {
+          reject(err);
+        } else {
+          resolve(data);
+        }
+      });
+    });
+  }
+
+  public sendResponse(id: RequestId, responseCode: number, result: ResponseBody): void {
+    const request = this.responses[id];
+    if (!request) {
+      throw new Error('No request found');
+    }
+    const {peer, method} = request;
+    const encodedResponse = encodeResponse(id, method, responseCode, result);
+    delete this.responses[id];
+    peer.write(encodedResponse);
+  }
+
+  public onRequestResponse(peer: Peer, data: Buffer): void {
+    if(!sanityCheckData(data)) {
+      // bad data
+      return;
+    }
+    // Changed response
+    let decodedBody, request;
+    try {
+      let decodedMessage = decodeMessage(data);
+      // only RPC requests/ responses should be passed here.
+      switch (decodedMessage.protocol) {
+        case ProtocolType.RPC:
+          request = deserialize(decodedMessage.payload, WireRequest);
+          decodedBody = decodeRequestBody(request.methodId, request.body);
+          break;
+        case ProtocolType.GOSSIP:
+          break;
+        case ProtocolType.PING:
+          break;
+      }
+    } catch (e) {
+      this.logger.warn('unable to decode request', e.message);
+      return;
+    }
+
+    const id = request.id;
+    const method = this.requests[id];
+    if (method === undefined) { // incoming request
+      this.responses[id] = {peer, method: request.method};
+      this.emit("request", peer.peerInfo, request.method, id, decodedBody);
+    } else { // incoming response
+      const event = `response ${id}`;
+      this.emit(event, null, decodedBody);
+    }
+  }
+
+  /**
+   * Connects to static peers
+   */
+  private async connectStaticPeers(): Promise<void> {
+    await Promise.all(this.bootnodes.map(async (bootnode: string): Promise<void> => {
+      let peerInfo = await hobbitsUriToPeerInfo(bootnode);
+      if (peerInfo) {
+        return this.dialForRpc(peerInfo);
+      }
+    }));
+  };
+
+  public async start(): Promise<void> {
+    this.wipDials = new Set();
+    this.peers = new Map<string, Peer>();
+
+    this.server.on("connection", async (connection): Promise<void> => {
+      const peerInfo = await socketConnectionToPeerInfo(connection);
+      this.addPeer(peerInfo);
+      this.logger.info(`Hobbits :: connected to: ${peerInfo.id.toB58String()}.`);
+    });
+
+    this.server.on("close", (): void => {
+      this.logger.info("Hobbits :: server closed.");
+      // this.running = false;
+    });
+
+    this.server.listen(HOBBITS_DEFAULT_PORT, (): void => {
+      this.logger.info("Hobbits :: server started.");
+      // this.running = true;
+    });
+
+    await this.connectStaticPeers();
+
+    // this.libp2p.handle(RPC_MULTICODEC, this.onConnection.bind(this));
```

Comment:
remove those comments

Reply:
I want to keep these here util issue #266 . They might help me.